api/handlers/products.js

```diff
@@ -0,0 +1,276 @@
+        return{
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 201,
+            body: JSON.stringify(result.rows[0])
+        }
+    }
+    } catch(error){
+        
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 500,
+            body: JSON.stringify({error: 'Internal server error'})
+        }
+
+    }
+}
+//update existing product
+module.exports.put = async (event) => {
+
+try{
+    
+    if (!event.body) {
+     return {
+        headers: {
+            'Access-Control-Allow-Origin': '*',
+           
+          },
+         statusCode: 400,
+         body: JSON.stringify({ msg: 'Please supply valid JSON data' })
+     };
+ } else{
+    const id = parseInt(event.pathParameters.id)
+    if(isNaN(id)){
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 400,
+            body: JSON.stringify({message: 'Invalid ID'}),
+        };
+    }
+    const requestBody = JSON.parse(event.body);
+    const {product_name, product_description, price, available_quantity}  =requestBody
+    const result = await pool.query('UPDATE products SET product_name=$1, product_description=$2, price=$3, available_quantity=$4 WHERE id=$5 RETURNING *', [product_name, product_description, price, available_quantity, id])
+    if(result.rows.length === 0){
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 404,
+            body: JSON.stringify({error: 'Product not found'})
+        }
+    } else{
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 200,
+            body: JSON.stringify(result.rows[0])
+        }
+    }
+
+ }
+    } catch(error){
+        return {
+            headers: {
+                'Access-Control-Allow-Origin': '*',
+               
+              },
+            statusCode: 500,
+            body: JSON.stringify({error: 'Internal server error'})
+        }
+
+    }
+}
+//delete existing product
+module.exports.delete = async (event) => {
+    try {
+        const id = parseInt(event.pathParameters.id)
+        if(isNaN(id)){
+            return {
+                headers: {
+                    'Access-Control-Allow-Origin': '*',
+                   
+                  },
+                statusCode: 400,
+                body: JSON.stringify({message: 'Invalid ID'}),
+            };
+        }
+        const result = await pool.query('DELETE FROM products WHERE id=$1 RETURNING *', [id]);
```

Comment:
There is a foreign key relationship in the database from `carts` to `products`, so if a product is in a cart it cannot be deleted and this SQL statement will fail. 

It might be that this is what you want—not letting products be deleted if they're in use—but if so it would be better to return an error saying that the product cannot be deleted because it's in a cart. Right now I think it would return "Internal server error".